### PR TITLE
Retry get json status when estimation sync time not calculated yet.

### PR DESCRIPTION
**Pull request by nick-wang**
Estimation sync time still need to be calculated based
on the synced data. Though less than a sec,
`drbdsetup status --json` will return an invalid json
format `"estimated-seconds-to-finish": nan,` when not ready.

### CODE REVIEW

**Comment by nick-wang** [comment]
Besides this change in drbd salt module, also working in progress on fix in drbd-utils. Though the later one will take longer time to upstream and back.

**Comment by nick-wang** [comment]
The fix in upstream https://github.com/LINBIT/drbd-utils/commit/104293030b2c0106b4791edb3eec38b476652a2e is the reason why this issue not shown in SLE15.

This commit doesn't need to be revert even the patch backported to SLE12SP4/SP5. Also this commit also helpful during the time of patch release in maintenance channel, which maybe quite long...

File: salt/modules/drbdmod.py [hunk 1]
```diff
@@ -280,6 +284,35 @@ def _is_no_backing_dev_request(res, output):
     return True
 
 
+def _get_json_output_save(command):
+    '''
```
Comment (arbulu89) [inline salt/modules/drbdmod.py:288]: Add the reason why we are creating this issue here in the docstring. I added some context in the main review comment

**Comment by arbulu89** [comment]
> The fix in upstream [LINBIT/drbd-utils@1042930](https://github.com/LINBIT/drbd-utils/commit/104293030b2c0106b4791edb3eec38b476652a2e) is the reason why this issue not shown in SLE15.
> 
> This commit doesn't need to be revert even the patch backported to SLE12SP4/SP5. Also this commit also helpful during the time of patch release in maintenance channel, which maybe quite long...

@nick-wang Is there any plan to backport this to SLE12? This would avoid having the need to have workarounds in our salt code

**Comment by nick-wang** [comment]
> @nick-wang Is there any plan to backport this to SLE12? This would avoid having the need to have workarounds in our salt code

@arbulu89 WIll backport the upstream fix to SLE12SP4/SP5, but still need this workaround before patch released.
Change the loop to context convert in the latest commit.

File: salt/modules/drbdmod.py [hunk 2]
```diff
@@ -280,6 +280,37 @@ def _is_no_backing_dev_request(res, output):
...
+    A warpper of get json command to acommandate json output issues
+    '''
+
+    error_str = '"estimated-seconds-to-finish": nan,'
+    replace_str = '"estimated-seconds-to-finish": 987654321,'
```
Comment (arbulu89) [inline salt/modules/drbdmod.py:289]: @nick-wang Why don't you convert the value to `null` ? In fact, `nan` is used for that. I think It makes sense to have as close as possible output as the final solution.

File: salt/modules/drbdmod.py [hunk 3]
```diff
@@ -280,6 +280,37 @@ def _is_no_backing_dev_request(res, output):
...
+
+    if 'retcode' not in results or results['retcode'] != 0:
+        LOGGER.info("Error running command \"%s\".  Error message: %s (%s)",
+                     command, results['stderr'], results['retcode'])
+        return None
+
+    # drbd-utils 104293030b2c0106b4791edb3eec38b476652a2e
+    # results['stdout'] is unicode
+    s_result = str(results['stdout'])
+    if error_str in s_result:
```
Comment (arbulu89) [inline salt/modules/drbdmod.py:300]: I think you can skip this `if` here. The `replace` will only take place if the 1st pattern is present. Anyway, It's fine as it is too.

File: salt/modules/drbdmod.py (hunk 2 above)
Comment (nick-wang) [reply salt/modules/drbdmod.py:289]: @arbulu89 I use `987654321` just match to the upstream change in https://github.com/LINBIT/drbd-utils/commit/104293030b2c0106b4791edb3eec38b476652a2e
, in which is use the number. 
Considering we use the function to check sync finished or not, the number is actually meaningless.

File: salt/modules/drbdmod.py (hunk 3 above)
Comment (nick-wang) [reply salt/modules/drbdmod.py:300]: Only use it to save/skip the step of `results['stdout'] = six.text_type(s_result)`.
It won't be performed in most cases, without a replacement, the results['stdout'] will keep the same.

**Pull request merged**